### PR TITLE
drastically improve docs

### LINKED ISSUE #135: Docs need a ton of cleanup
A few things I have complaints about:
* [x] Bulleted lists are hard to traverse
* [x] Examples for sourcemaps ended up as bulleted list
* [x] Order of docs should be improved
  * [x] Mention of `dest` updating the file objects when passed through should be at the top - ref #76 
* [x] What is vinyl, vinyl adapter

### PR DESCRIPTION
**Pull request by phated**
Fixes #135 

@contra please review thoroughly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,153 +1,277 @@
-# vinyl-fs [![NPM version][npm-image]][npm-url] [![Build Status][travis-image]][travis-url] [![Coveralls Status][coveralls-image]][coveralls-url] [![Dependency Status][depstat-image]][depstat-url]
-## Information
-<table>
-  <tr><td>Package</td><td>vinyl-fs</td></tr>
-  <tr><td>Description</td><td>Vinyl adapter for the file system</td></tr>
-  <tr><td>Node Version</td><td>>= 0.10</td></tr>
-</table>
+<p align="center">
```
Comment (contra) [inline README.md]: dont think we need the gulp logo since this library is used by other tools

File: README.md [hunk 2]
```diff
@@ -1,153 +1,277 @@
...
+  <a href="http://gulpjs.com">
+    <img height="257" width="114" src="https://raw.githubusercontent.com/gulpjs/artwork/master/gulp-2x.png">
+  </a>
+</p>
+
+# vinyl-fs
+
+[![NPM version][npm-image]][npm-url] [![Downloads][downloads-image]][npm-url] [![Build Status][travis-image]][travis-url] [![Coveralls Status][coveralls-image]][coveralls-url] [![Gitter chat][gitter-image]][gitter-url]
+
+Vinyl adapter for the file system.
```
Comment (contra) [inline README.md]: should link to the vinyl project

File: README.md [hunk 3]
```diff
@@ -1,153 +1,277 @@
...
+
+## What is Vinyl?
+
+Vinyl is a very simple metadata object that describes a file. When you think of a file, two attributes come to mind: `path` and `contents`. These are the main attributes on a Vinyl object. A file does not necessarily represent something on your computer’s file system. You have files on S3, FTP, Dropbox, Box, CloudThingly.io and other services. Vinyl can be used to describe files from all of these sources.
```
Comment (contra) [inline README.md]: link to vinyl project

File: README.md [hunk 4]
```diff
@@ -1,153 +1,277 @@
...
+
+## What is a Vinyl Adapter?
+
+While Vinyl provides a clean way to describe a file, we now need a way to access these files. Each file source needs what I call a "Vinyl adapter". A Vinyl adapter simply exposes a `src(globs)` and a `dest(folder)` method.  Each return a stream.  The `src` stream produces Vinyl objects, and the `dest` stream consumes Vinyl objects.  Vinyl adapters can expose extra methods that might be specific to their input/output medium, such as the `symlink` method `vinyl-fs` provides.
```
Comment (contra) [inline README.md]: some of these sentences have multiple spaces in between

File: README.md (hunk 1 above)
Comment (phated) [inline README.md]: For consistency, this is being added to all gulpjs projects to indicate it follows our eslint/jscs, testing standards and exists in the gulpjs org

File: README.md (hunk 4 above)
Comment (phated) [inline README.md]: I think that markdown removes the extra spaces, but I can remove

**Comment by phated** [comment]
Updated

File: README.md (hunk 1 above)
Comment (contra) [inline README.md]: @phated makes sense

**Comment by phated** [comment]
Alright.  Going to :shipit:

**Pull request merged**